### PR TITLE
Update dependencies and workaround ARM64 test failure

### PR DESCRIPTION
**Pull request by robmen**
* Update many NuGet packages
* Clean up shipping WiX .nupkgs
* Workaround ARM64 test failure until there is an ARM64 NETFX wix.exe - related to wixtoolset/issues#6960

### CODE REVIEW

File: src/internal/SetBuildNumber/Directory.Packages.props.pp [hunk 1]
```diff
@@ -45,18 +45,18 @@
...
     <PackageVersion Include="System.Reflection.Metadata" Version="1.6.0" />
     <PackageVersion Include="System.Security.Principal.Windows" Version="4.7.0" />
-    <PackageVersion Include="System.Text.Encoding.CodePages" Version="4.6.0" />
+    <PackageVersion Include="System.Text.Encoding.CodePages" Version="4.7.0" />
     <PackageVersion Include="System.Text.Json" Version="4.6.0" />
 
     <PackageVersion Include="Microsoft.AspNetCore.Owin" Version="3.1.13" />
     <PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="3.8.0" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.3.1" />
```
Comment (rseanhall) [inline src/internal/SetBuildNumber/Directory.Packages.props.pp:56]: Is this necessary? I vaguely remember needing to keep this lower to support VS2019, but that might not be correct.

File: src/internal/SetBuildNumber/Directory.Packages.props.pp [hunk 2]
```diff
@@ -45,18 +45,18 @@
...
     <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing.XUnit" Version="1.1.1" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.8.0" />
-    <PackageVersion Include="Microsoft.VisualStudio.Setup.Configuration.Native" Version="1.14.114" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.3.1" />
```
Comment (rseanhall) [inline src/internal/SetBuildNumber/Directory.Packages.props.pp:58]: Same as above.

File: src/internal/SetBuildNumber/Directory.Packages.props.pp (hunk 1 above)
Comment (rseanhall) [reply src/internal/SetBuildNumber/Directory.Packages.props.pp:56]: This change will require users of WixToolset.Dnc.HostGenerator to have VS2022 17.3 or later - https://github.com/dotnet/roslyn/blob/main/docs/wiki/NuGet-packages.md.

Comment (robmen) [reply src/internal/SetBuildNumber/Directory.Packages.props.pp:56]: Not strictly necessary. Is it necessary to keep it at `3.8.0` for VS2019 support?

Comment (rseanhall) [reply src/internal/SetBuildNumber/Directory.Packages.props.pp:56]: According to that page, it needs to be 3.x to support VS2019.

Comment (rseanhall) [reply src/internal/SetBuildNumber/Directory.Packages.props.pp:56]: It looks like VS2019 16.11 is the only supported version after this month so we should be able to go to `3.11.0`.

Comment (robmen) [reply src/internal/SetBuildNumber/Directory.Packages.props.pp:56]: Weird. Your second comment didn't show up until I refreshed the page. That's a good doc link.

Comment (robmen) [reply src/internal/SetBuildNumber/Directory.Packages.props.pp]: Updated to 3.11 and added the link to that documentation.

Comment (rseanhall) [reply src/internal/SetBuildNumber/Directory.Packages.props.pp]: Looks good.

**Pull request merged**